cfme/services/catalogs/orchestration_template.py

```diff
@@ -1,166 +1,284 @@
-                ]
-            }
-        ]
-    }
-)
-
-
-class OrchestrationTemplate(Updateable, Pretty):
-    pretty_attrs = ['template_type', 'template_name']
-
-    def __init__(self, template_type=None, template_name=None, description=None):
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+from widgetastic.widget import Text, Checkbox
+from widgetastic_patternfly import BootstrapSelect, Button, Input
+from widgetastic_manageiq import ScriptBox
+
+from . import ServicesCatalogView
+
+
+class OrchestrationTemplatesAllView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            super(OrchestrationTemplatesAllView, self).is_displayed and
+            self.title.text == 'All Orchestration Templates' and
+            self.orchestration_templates.is_opened and
+            self.orchestration_templates.tree.currently_selected == ["All Orchestration Templates"])
+
+
+class TemplateForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    description = Input(name="description")
+    template_type = BootstrapSelect("type")
+    draft = Checkbox(name='draft')
+    content = ScriptBox(locator="//div[@id='basic_info_div']/div/div"
+                    "/div/div/div/div/div/div/pre/span")
+
+    cancel_button = Button('Cancel')
+
+
+class CopyTemplateForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    description = Input(name="description")
+    draft = Checkbox(name='draft')
+    content = ScriptBox(locator="//div[@id='basic_info_div']/div/div"
+                    "/div/div/div/div/div/div/pre/span")
+
+    cancel_button = Button('Cancel')
+
+
+class AddTemplateView(TemplateForm):
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Adding a new Orchestration Template" and
+            self.orchestration_templates.is_opened
+        )
+
+
+class EditTemplateView(TemplateForm):
+
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Editing {}".format(self.obj.name) and
+            self.orchestration_templates.is_opened
+        )
+
+
+class CopyTemplateView(CopyTemplateForm):
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.is_displayed and
+            self.title.text == "Copying {}".format(self.obj.name) and
+            self.orchestration_templates.is_opened
+        )
+
+
+class DetailsTemplateView(TemplateForm):
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == '{} "{}"'.format(self.obj.template_type[:-1],
```

Comment:
For example, template_type == 'CloudFormation Templates', but title is "CloudFormation Template "ASDasdwe" ' (without "s" in the end)